### DISCUSSION TITLE
Quarkus Hibernate using Oracle UCP (Universal Connection Pool)

### DISCUSSION THREAD

**Discussion by fdoyledafm** [quarkusio/quarkus#43494] [Q&A]
Hello,

Is it possible to use a UCP connection with quarkus-hibernate-orm?

I'm using Quarkus 3.14

Thanks

[a comment by an automated account is not shown]

**Comment by yrodiere** [comment]
Hello,

It's not been evaluated by the Quarkus team, and there is no code or tests related to it in Quarkus.

That doesn't mean it cannot work, but that does mean you'll have to make it work yourself. And there's a significant chance it won't work in native images by default.

That being said, Quarkus ships with the Agroal connection pool, and uses it in Hibernate ORM, so [connection pooling is already enabled by default](https://quarkus.io/guides/datasource#jdbc-connection-pool-size-adjustment). I'm not familiar with UCP, so I'm not sure what it brings to the table compared to a DB-agnostic pool such as Agroal.

**Reply by yrodiere** [reply]
FWIW I tried to use UCP in Quarkus and at least the basics seem to work: https://github.com/yrodiere/quarkus-playground/tree/oracle-ucp

Though native compilation, of course, does not.

Still, since nobody is putting in the work of actually maintaining an integration and testing in Quarkus itself, use this at your own risk :)
